state/protocol/badger/mutator_test.go

```diff
@@ -143,7 +143,13 @@ func TestSealedIndex(t *testing.T) {
 		require.NoError(t, err)
 
 		// block 2(result B1)
+		result, _, err := rootSnapshot.SealedResult()
+		require.NoError(t, err)
+
+		serviceEvent := result.ServiceEvents[0]
+
 		b1Receipt := unittest.ReceiptForBlockFixture(b1)
+		b1Receipt.ExecutionResult.ServiceEvents = []flow.ServiceEvent{serviceEvent}
```

Comment:
I would say that this event being invalid is even better for the purpose - since the index is supposed to simply index events, not consider their type and then relevance

Reply:
> index is supposed to simply index events, not consider their type and then relevance

That's a good point. I was thinking that we would not want to index invalid service events, but I can see why we would want to do that. Though it will mean that, the code querying this index might need to be able to handle the case where querying the latest event results in an event that was considered invalid when it was processed.

Reply:
Yes, I would assume the code using index must be aware of the possibility